src/Tests/Microsoft.Diagnostics.Monitoring.StartupHook.UnitTests/Exceptions/Identification/ExceptionIdentifierCacheTests.cs

```diff
@@ -0,0 +1,192 @@
+
+            // Validate exception type token in cache
+            Assert.NotEqual(InvalidId, exceptionClassData.Token);
+            uint parentClassToken = ValidateTokenAndGetOuterToken(
+                callback.NameCache,
+                exceptionClassData.ModuleId,
+                exceptionClassData.Token,
+                ex.GetType());
+
+            // Validate exception type parent token in cache
+            Assert.NotEqual(InvalidId, parentClassToken);
+            parentClassToken = ValidateTokenAndGetOuterToken(
+                callback.NameCache,
+                exceptionClassData.ModuleId,
+                parentClassToken,
+                ex.GetType().DeclaringType!);
+            Assert.Equal(InvalidId, parentClassToken);
+
+            // Validate exception type generic argument in cache
+            ulong genericArgClassId = Assert.Single(exceptionClassData.TypeArgs);
+            Assert.True(callback.NameCache.ClassData.TryGetValue(genericArgClassId, out ClassData? genericArgClassData));
+            Assert.NotNull(genericArgClassData);
+
+            // Validate exception type generic argument token in cache
+            Assert.NotEqual(InvalidId, genericArgClassData.Token);
+            parentClassToken = ValidateTokenAndGetOuterToken(
+                callback.NameCache,
+                genericArgClassData.ModuleId,
+                genericArgClassData.Token,
+                typeof(int));
+            Assert.Equal(InvalidId, parentClassToken);
+
+            // Validate throwing method in cache
+            Assert.NotEqual(InvalidId, exceptionIdData.ThrowingMethodId);
+            Assert.True(callback.NameCache.FunctionData.TryGetValue(exceptionIdData.ThrowingMethodId, out FunctionData? throwingMethodData));
+            Assert.NotNull(throwingMethodData);
+
+            // Validate throwing method module in cache
+            Assert.NotEqual(InvalidId, throwingMethodData.ModuleId);
+            Assert.True(callback.NameCache.ModuleData.TryGetValue(throwingMethodData.ModuleId, out ModuleData? throwingMethodModuleData));
+            Assert.NotNull(throwingMethodModuleData);
+            Assert.Equal(ThisModuleName, throwingMethodModuleData.Name);
+
+            // Validate throwing method remaining properties
+            Assert.Equal(nameof(ExceptionIdentifierCache_ThrownException), throwingMethodData.Name);
+            Assert.NotEqual(InvalidId, throwingMethodData.ParentClass);
+            Assert.NotEqual(InvalidToken, throwingMethodData.ParentToken);
+            Assert.Empty(throwingMethodData.TypeArgs);
+        }
+
+        private static uint ValidateTokenAndGetOuterToken(NameCache cache, ulong moduleId, uint token, Type expectedType)
+        {
+            Assert.True(cache.TokenData.TryGetValue(new ModuleScopedToken(moduleId, token), out TokenData? tokenData));
+            Assert.NotNull(tokenData);
+            string? expectedTypeName = null == expectedType.DeclaringType ?
```

Comment:
This is technically slightly different than the way this is implemented earlier (`null == expectedType.DeclaringType ? expectedType.FullName ?? expectedType.Name : expectedType.Name`) - should this be standardized for consistency?

Reply:
I don't see anywhere else in the codebase that has `null == expectedType.DeclaringType`; can you point me directly to what you are referencing?

Reply:
That was in reference to this: https://github.com/dotnet/dotnet-monitor/pull/3406/files#diff-aea088e1c05d63b61267f1dadf91ab9b32525969da9b1f6a272dd9934079d09fR160 - it's possible I just noticed they looked similar but aren't meant to behave the same way.

Reply:
These don't need to be the same. The product code is handling all cases where `Type.FullName` _may be_ null (https://learn.microsoft.com/en-us/dotnet/api/system.type.fullname?view=net-7.0#property-value) whereas the test code knows its a fully resolved type that is an Exception type.